server/protocols/jmap/src/main/java/org/apache/james/jmap/utils/MailboxUtils.java

```diff
@@ -61,10 +61,10 @@ public MailboxUtils(MailboxManager mailboxManager, MailboxMapperFactory mailboxM
     public Optional<Mailbox> mailboxFromMailboxPath(MailboxPath mailboxPath, MailboxSession mailboxSession) {
         try {
             Optional<Role> role = Role.from(mailboxPath.getName());
-            MessageManager.MetaData mailboxMetaData = getMailboxMetaData(mailboxPath, mailboxSession);
-            String mailboxId = getMailboxId(mailboxPath, mailboxSession);
+            MessageManager mailbox = mailboxManager.getMailbox(mailboxPath, mailboxSession);
+            MessageManager.MetaData mailboxMetaData = getMailboxMetaData(mailbox, mailboxSession);
             return Optional.ofNullable(Mailbox.builder()
-                    .id(mailboxId)
+                    .id(mailbox.getId().serialize())
```

Comment:
Because Mailbox will be converted into JSON. How do you tell Jackson how to serialize MailboxId else?